src/main/utils/get-installed-app-ids.ts

```diff
@@ -12,54 +13,115 @@ import { dispatch } from '../state/store'
 
 const execFileP = promisify(execFile)
 
-async function getAllInstalledBundleIds(): Promise<string[]> {
+async function getAllInstalledAppBundleIds(): Promise<[string[], string[]]> {
   const { stdout: allApps } = await execFileP('mdfind', [
-    "kMDItemKind == 'Application'",
     '-onlyin',
     '/Applications',
     '-onlyin',
     path.join(homedir(), 'Applications'),
+    "kMDItemKind == '*'",
+    '-attr',
+    'kMDItemCFBundleIdentifier',
   ])
 
-  const bundleIds = await Promise.all(
-    allApps
-      .trim()
-      .split('\n')
-      .map(async (entry) => {
-        const { stdout: bundleId } = await execFileP('mdls', [
-          '-name',
-          'kMDItemCFBundleIdentifier',
-          '-raw',
-          entry,
-        ])
-
-        return bundleId
-      }),
-  )
-
-  return bundleIds
-}
+  const appPaths: string[] = []
+  const bundleIds: string[] = []
+  const appsList = allApps.trim().split('\n')
 
-async function getInstalledAppIds(): Promise<void> {
-  dispatch(startedScanning())
+  for (const result of appsList) {
+    const pathAndId = result.split('   ')
 
-  const allInstalledBundleIds = await getAllInstalledBundleIds()
-  const installedApps: AppId[] = []
+    if (pathAndId.length === 2) {
+      const appID = pathAndId[1].split(' = ')
 
-  for (const installedBundleId of allInstalledBundleIds) {
-    if (installedBundleId in apps) {
-      installedApps.push(installedBundleId as AppId)
+      if (appID.length === 2 && !appID[1].includes('null')) {
+        appPaths.push(pathAndId[0])
+        bundleIds.push(appID[1])
+      }
     }
   }
 
-  // It appears that sometimes the installed app IDs are not fetched, maybe a
-  // race with Spotlight index? So if none found, keep retrying.
-  // https://github.com/will-stone/browserosaurus/issues/425
-  if (installedApps.length === 0) {
-    await sleep(500)
-    getInstalledAppIds()
+  return [appPaths, bundleIds]
+}
+
+async function getAllInstalledApps(): Promise<string[]> {
+  const hasUserAppsFolder = fs.existsSync(path.join(homedir(), 'Applications'))
+  let findArguments: string[]
+
+  if (hasUserAppsFolder) {
+    findArguments = [
+      '/Applications',
+      path.join(homedir(), 'Applications'),
+      '-iname',
+      '*.app',
+      '-maxdepth',
+      '1',
```

Comment:
Ah, this doesn't work if there are folders in `/Applications`. I'm going to release a small patch (not using `find`), and I'll post on the issue when it's ready.

Reply:
I have the standard 'Utilities' folder in my `/Applications` folder but it is excluded based on the '-iname- argument. Can you elaborate on what exactly is not working for you?